src/mbgl/renderer/sources/render_tile_source.cpp

```diff
@@ -19,15 +21,23 @@ bool RenderTileSource::isLoaded() const {
 }
 
 void RenderTileSource::upload(gfx::UploadPass& parameters) {
-    tilePyramid.upload(parameters);
+    for (auto& tile : renderTiles) {
+        tile.upload(parameters);
+    }
 }
 
 void RenderTileSource::prepare(const SourcePrepareParameters& parameters) {
-    tilePyramid.prepare(parameters);
+    renderTiles.clear();
+    for (auto& entry : tilePyramid.getRenderTiles()) {
+        renderTiles.emplace_back(entry.first, entry.second);
```

Comment:
is it possible to preallocate this, or enumerate the same list without reallocation, or use alternative data structure? Is the container emplace back introduced by this patch or refactored?

Reply:
can call `reserve` beforehand